### PR TITLE
contracts: mapping for finalization time per target chain id

### PR DESCRIPTION
**Pull request by fredo**
introduce a mapping from for finalization times per chain id.
If not contained target chain id is not supported by the protocol.

### CODE REVIEW

File: scripts/deployment/main.py [hunk 1]
```diff
@@ -89,7 +89,9 @@ def deploy_l1(web3: Web3) -> dict[str, DeployedContract]:
     return {"Resolver": resolver}
 
 
-def deploy_l2(web3: Web3, resolver: Contract) -> dict[str, DeployedContract]:
+def deploy_l2(
+    web3: Web3, resolver: Contract, supported_target_chains: list[tuple[int, int]]
```
Comment (palango) [inline scripts/deployment/main.py:93]: why not use a simple dict here?

File: contracts/contracts/RequestManager.sol [hunk 2]
```diff
@@ -391,4 +393,9 @@ contract RequestManager is Ownable {
         gasPrice = newGasPrice;
         serviceFeePPM = newServiceFeePPM;
     }
+
+    function addTargetChain(uint256 targetChainId, uint256 finalizationTime) external onlyOwner {
+        require(finalizationTime > 0, "Finalization time of roll up must be greater 0");
```
Comment (istankovic) [inline contracts/contracts/RequestManager.sol:398]: Nitpick: roll up -> rollup

File: contracts/contracts/RequestManager.sol [hunk 3]
```diff
@@ -391,4 +393,9 @@ contract RequestManager is Ownable {
...
+        finalizationTimes[targetChainId] = finalizationTime;
+    }
```
Comment (istankovic) [inline contracts/contracts/RequestManager.sol:400]: As this function can be called multiple times for the same chain ID, I would rename it to something like `setFinalizationTime`.

File: contracts/contracts/RequestManager.sol [hunk 4]
```diff
@@ -391,4 +392,9 @@ contract RequestManager is Ownable {
         gasPrice = newGasPrice;
         serviceFeePPM = newServiceFeePPM;
     }
+
+    function setFinalizationTime(uint256 targetChainId, uint256 finalizationTime) external onlyOwner {
+        require(finalizationTime > 0, "Finalization time of rollup must be greater than 0");
```
Comment (palango) [inline contracts/contracts/RequestManager.sol:397]: Maybe something like "Rollup is not registered" is better? The current error is not really actionable for the user.

Comment (fredo) [reply contracts/contracts/RequestManager.sol:397]: Isn't the only user the owner of the contract, hence we?

**Pull request merged**